core/src/internal/event_system.c

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#ifdef _CRTDBG_MAP_ALLOC
+#include <crtdbg.h>
+#endif
+
+#include "azure_c_shared_utility/vector.h"
+#include "azure_c_shared_utility/gballoc.h"
+#include "azure_c_shared_utility/xlogging.h"
+#include "azure_c_shared_utility/threadapi.h"
+#include "azure_c_shared_utility/lock.h"
+#include "azure_c_shared_utility/condition.h"
+#include "azure_c_shared_utility/list.h"
+
+#include "gateway_ll.h"
+#include "internal/event_system.h"
+
+#include <assert.h>
+#include <stdlib.h>
+
+struct EVENTSYSTEM_DATA {
```

Comment:
@avranju No need, it's internal and the struct name itself is not anywhere inside the code